internal/pam/pam_test.go

```diff
@@ -31,21 +31,24 @@ func TestAuthenticate(t *testing.T) {
+		"authenticate successfully with unmatched case (online)":                  {username: "Success@Domain.COM"},
+		"authenticate successfully (online) with offline authentication disabled": {username: "success@domain.com"},
 
 		// error cases
-		"error on invalid conf":                               {conf: "invalid-aad.conf", wantErrType: pam.ErrPamSystem},
-		"error on unexisting conf":                            {conf: "doesnotexist.conf", wantErrType: pam.ErrPamSystem},
-		"error on unexisting users":                           {username: "no such user", wantErrType: pam.ErrPamAuth},
-		"error on invalid password":                           {username: "invalid credentials", wantErrType: pam.ErrPamAuth},
-		"error on offline with user online user not in cache": {conf: "forceoffline.conf", initialCache: "db_with_expired_users", wantErrType: pam.ErrPamAuth},
-		"error on offline with purged user account":           {username: "purgeduser@domain.com", initialCache: "db_with_expired_users", wantErrType: pam.ErrPamAuth},
-		"error on offline with expired user account":          {conf: "forceoffline.conf", initialCache: "db_with_expired_users", username: "expireduser@domain.com", wantErrType: pam.ErrPamAuth},
-		"error on offline with unpurged old user account":     {conf: "forceoffline-expire-right-away.conf", initialCache: "db_with_expired_users", username: "purgeduser@domain.com", wantErrType: pam.ErrPamAuth},
-		"error on server error":                               {username: "unreadable server response", wantErrType: pam.ErrPamAuth},
-		"error on cache can't be created/opened":              {wrongCacheOwnership: true, wantErrType: pam.ErrPamSystem},
+		"error on invalid conf":                                 {conf: "invalid-aad.conf", wantErrType: pam.ErrPamSystem},
+		"error on unexisting conf":                              {conf: "doesnotexist.conf", wantErrType: pam.ErrPamSystem},
+		"error on unexisting users":                             {username: "no such user", wantErrType: pam.ErrPamAuth},
+		"error on invalid password":                             {username: "invalid credentials", wantErrType: pam.ErrPamAuth},
+		"error on offline with user online user not in cache":   {conf: "forceoffline.conf", initialCache: "db_with_expired_users", wantErrType: pam.ErrPamAuth},
+		"error on offline with expired user":                    {conf: "forceoffline.conf", initialCache: "db_with_expired_users", username: "expireduser@domain.com", wantErrType: pam.ErrPamAuth},
+		"error on offline with purged user":                     {conf: "forceoffline-expire-right-away.conf", initialCache: "db_with_expired_users", username: "purgeduser@domain.com", wantErrType: pam.ErrPamAuth},
+		"error on offline with offline authentication disabled": {conf: "forceoffline-offline-auth-disabled.conf", initialCache: "users_in_db", username: "myuser@domain.com", wantErrType: pam.ErrPamAuth},
```

Comment:
same comment that the other comment on the removed tests which is slightly different, even if the same code path.